RPC test deadlocks when ledger channels are created concurrently

Running TestRpcClient locally on https://github.com/statechannels/go-nitro/commit/73b2fe854b29670a459f105b7c9f5f4181878a10 results in:

```
--- FAIL: TestRpcClient (10.07s)
    helpers_test.go:69: Objective ids [DirectFunding-0xee2c00258f3283d405f19fcd9e9abf0eb6ce992f5ba0dca1387075d2212414cc] failed to complete on client 0x111A00868581f73AB42FEEF67D235Ca09ca1E8db within 10s
FAIL
exit status 1
```

--- a/client_test/rpc_test.go
+++ b/client_test/rpc_test.go
@@ -67,9 +67,8 @@ func TestRpcClient(t *testing.T) {
 	assert.Regexp(t, "DirectFunding.0x.*", res.Id)
 
 	waitTimeForCompletedObjectiveIds(t, &clientA, defaultTimeout, res.Id)
-	waitTimeForCompletedObjectiveIds(t, &clientI, defaultTimeout, res.Id)
 	waitTimeForCompletedObjectiveIds(t, &clientB, defaultTimeout, bobResponse.Id)
-	waitTimeForCompletedObjectiveIds(t, &clientI, defaultTimeout, bobResponse.Id)
+	waitTimeForCompletedObjectiveIds(t, &clientI, defaultTimeout, res.Id, bobResponse.Id)
 
 	vRes := rpcClientA.CreateVirtual(
 		[]types.Address{irene.Address()},

--- a/client_test/rpc_test.go
+++ b/client_test/rpc_test.go
@@ -61,14 +61,13 @@ func TestRpcClient(t *testing.T) {
 	defer rpcClientA.Close()
 
 	res := rpcClientA.CreateLedger(irene.Address(), 100, testdata.Outcomes.Create(alice.Address(), irene.Address(), 100, 100, types.Address{}))
+	bobResponse := clientB.CreateLedgerChannel(irene.Address(), 100, testdata.Outcomes.Create(bob.Address(), irene.Address(), 100, 100, types.Address{}))
 
 	// Quick sanity check that we're getting a valid objective id
 	assert.Regexp(t, "DirectFunding.0x.*", res.Id)
+
 	waitTimeForCompletedObjectiveIds(t, &clientA, defaultTimeout, res.Id)
 	waitTimeForCompletedObjectiveIds(t, &clientI, defaultTimeout, res.Id)
-
-	bobResponse := clientB.CreateLedgerChannel(irene.Address(), 100, testdata.Outcomes.Create(bob.Address(), irene.Address(), 100, 100, types.Address{}))
-
 	waitTimeForCompletedObjectiveIds(t, &clientB, defaultTimeout, bobResponse.Id)
 	waitTimeForCompletedObjectiveIds(t, &clientI, defaultTimeout, bobResponse.Id)
 

